src/components/UI/Display/Apps/AppList/AppGrid.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { memo, useLayoutEffect, useRef, useState } from 'react';
+import { WindowScroller } from 'react-virtualized';
+import {
+  areEqual,
+  GridChildComponentProps,
+  VariableSizeGrid,
+} from 'react-window';
+import styled from 'styled-components';
+import { IAppProps } from 'UI/Display/Apps/AppList/App';
+
+interface IAppGrid {
+  items: IAppProps[];
+  itemMinWidth: number;
+  itemMinHeight: number;
+  gridGap: number;
+  render: (item: IAppProps, i: number) => React.ReactNode;
+}
+
+interface IScrollPosition {
+  scrollTop: number;
+}
+
+interface IDataProps {
+  items: IAppProps[];
+  columnCount: number;
+  gridGap: number;
+  render: (item: IAppProps, i: number) => React.ReactNode;
+}
+
+const StyledContainer = styled.div<{ gridGap: number }>`
+  margin: 0 -${(props) => props.gridGap / 2}px;
+`;
+
+const StyledGrid = styled(VariableSizeGrid)<{ minHeight: number }>`
+  height: 100% !important;
+
+  > div {
+    height: ${(props) => props.minHeight}px !important;
+  }
```

Comment:
The height for the outer `div` element has to be overwritten to allow it to be scrolled by scrolling the page (otherwise it would be a scrollable window within the page).

The height for the inner `div` element of the virtualized list increases as the user scrolls. This presents a problem when navigating back to this page because the scroll position has a larger number than the initial height of the element, and it will not restore the correct scroll position. Setting the height to be the total height of the elements ensures that the page will have a greater height than the scroll position.